app/drivers/zephyr/kscan_gpio_demux.c

```diff
@@ -0,0 +1,270 @@
+        struct kscan_gpio_irq_callback_##n *data =                                                 \
+            CONTAINER_OF(cb, struct kscan_gpio_irq_callback_##n, callback);                        \
+        CHECK_DEBOUNCE_CFG(n, ({ k_work_submit(data->work); }), ({                                 \
+                               k_delayed_work_cancel(data->work);                                  \
+                               k_delayed_work_submit(data->work,                                   \
+                                                     K_MSEC(DT_INST_PROP(n, debounce_period)));    \
+                           }))                                                                     \
+    }                                                                                              \
+                                                                                                   \
+    static struct kscan_gpio_data_##n kscan_gpio_data_##n = {                                      \
+        .rows = {[INST_MATRIX_INPUTS(n) - 1] = NULL}, .cols = {[INST_DEMUX_GPIOS(n) - 1] = NULL}}; \
+                                                                                                   \
+    /* KSCAN API configure function */                                                             \
+    static int kscan_gpio_configure_##n(struct device *dev, kscan_callback_t callback) {           \
+        LOG_DBG("KSCAN API configure");                                                            \
+        struct kscan_gpio_data_##n *data = dev->driver_data;                                       \
+        if (!callback) {                                                                           \
+            return -EINVAL;                                                                        \
+        }                                                                                          \
+        data->callback = callback;                                                                 \
+        LOG_DBG("Configured GPIO %d", n);                                                          \
+        return 0;                                                                                  \
+    };                                                                                             \
+                                                                                                   \
+    /* KSCAN API enable function */                                                                \
+    static int kscan_gpio_enable_##n(struct device *dev) {                                         \
+        LOG_DBG("KSCAN API enable");                                                               \
+        struct kscan_gpio_data_##n *data = dev->driver_data;                                       \
+        k_timer_start(&data->poll_timer, K_MSEC(POLL_INTERVAL(n)), K_MSEC(POLL_INTERVAL(n)));      \
```

Comment:
I believe any running timers can possibly wake Zephyr from deep sleep on nrf52. We don't need to solve in this PR, but we might want a follow up to hook into the sleep state hooks in Zephyr, and disable this timer when we enter a sleep state.